lib/rules/no-hooks-from-ancestor-modules.js

```diff
@@ -0,0 +1,114 @@
+/**
+ * @fileoverview forbid the use of hooks from ancestor modules
+ * @author Raymond Cohen
+ */
+"use strict";
+
+//------------------------------------------------------------------------------
+// Requirements
+//------------------------------------------------------------------------------
+
+const utils = require("../utils");
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+const NESTABLE_HOOK_NAMES = ["afterEach", "beforeEach"];
+
+module.exports = {
+    meta: {
+        docs: {
+            description: "forbid the use of hooks from ancestor modules",
+            category: "Possible Errors",
+            recommended: false
+        },
+        fixable: null,
+        messages: {
+            "noHooksFromAncestorModules": "Do not use hooks from an ancestor module. You called {{usedHookIdentifierName}}.{{hookName}} inside module({{containingModuleDescription}})."
```

Comment:
I love the greater level of detail here! ...Now, unfortunately, I need to ask if we can shorten the message. 😅 

Let me know if you need help thinking of how you could shorten the message!

Reply:
I could drop the "inside module(...)" portion, or are you looking for even shorter?

Reply:
Even shorter might be better! If we want to mention the module's description, maybe we could do something like this?

```suggestion
            "noHooksFromAncestorModules": "Do not use `{{usedHookIdentifierName}}.{{hookName}}` from module({{containingModuleDescription}}) inside nested module."
```

Reply:
I've shortened the message even more - since you'll see where the warning is, there's probably no extra value to the description of the containing module.